### PR TITLE
Part of #18714 and #17670: Replacing deprecated constants

### PR DESCRIPTION
**Pull request by dhruvv173**
## Explanation
This PR is a part of the following issues

- #18714 
- #17670 

Made changes to follow files
`ui/components/app/snaps/copyable/copyable.js`
`ui/components/app/snaps/snap-authorship-expanded/snap-authorship-expanded.js`
`ui/components/app/snaps/snap-version/snap-version.js`

### Story for `Copyable`
image
image

### Story for `SnapAuthorshipExpanded`
image
image

### Before
image
### After
image

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by dhruvv173** [comment]
hello @georgewrmarshall @garrettbear 
I'm really sorry for closing the previous [PR](https://github.com/MetaMask/metamask-extension/pull/19519), I messed up 😢 
Can you please rereview this? Thanks!

File: ui/components/app/snaps/copyable/copyable.js [hunk 1]
```diff
@@ -1,26 +1,33 @@
...
+  Text,
+  Box,
+} from '../../../component-library';
 
 export const Copyable = ({ text }) => {
   const [copied, handleCopy] = useCopyToClipboard();
   return (
     <Box
       className="copyable"
+      display={Display.Flex}
```
Comment (dhruvv173) [inline ui/components/app/snaps/copyable/copyable.js:30]: before
image
after
image

File: ui/components/app/snaps/snap-version/snap-version.js [hunk 2]
```diff
@@ -29,6 +30,7 @@ const SnapVersion = ({ version, url }) => {
     >
       <Box
         className="snap-version__wrapper"
+        display={Display.Flex}
```
Comment (dhruvv173) [inline ui/components/app/snaps/snap-version/snap-version.js:33]: before
image
after
image

**Review by garrettbear — DISMISSED** [review]
Looks good. @georgewrmarshall we should coordinate on helping fix that `Copyable` component to be either `as="button"` or having them use the `ButtonIcon` (preferred) method. 

I tested it with the `as="button"` and it appears to work, but is jumping 1px so I didn't want to suggest it for this PR.

File: ui/components/app/app-components.scss [hunk 3]
```diff
@@ -35,8 +35,6 @@
 @import 'snaps/snap-ui-markdown/index';
 @import 'snaps/snap-delineator/index';
 @import 'snaps/snap-settings-card/index';
-@import 'snaps/copyable/index';
-@import 'snaps/snap-version/index';
 @import 'gas-details-item/index';
```
Comment (georgewrmarshall) [inline ui/components/app/app-components.scss:38]: Removing some more unused CSS

File: ui/components/app/snaps/copyable/copyable.js (hunk 1 above)
Comment (georgewrmarshall) [reply ui/components/app/snaps/copyable/copyable.js:23]: Made some updates so it uses the `ButtonIcon` component

### After


https://github.com/MetaMask/metamask-extension/assets/8112138/0aecd1e7-70b4-4418-8e69-b67f7792b740

File: ui/components/app/snaps/snap-authorship-expanded/snap-authorship-expanded.js [hunk 4]
```diff
@@ -163,14 +169,12 @@ const SnapAuthorshipExpanded = ({ snapId, className, snap }) => {
...
             alignItems={AlignItems.center}
-            marginTop={4}
           >
-            <Text variant={TextVariant.bodyMd} fontWeight={FontWeight.Medium}>
-              {t('version')}
-            </Text>
+            <Text variant={TextVariant.bodyMdMedium}>{t('version')}</Text>
             <SnapVersion version={snap?.version} url={url} />
           </Box>
         </Box>
```
Comment (georgewrmarshall) [inline ui/components/app/snaps/snap-authorship-expanded/snap-authorship-expanded.js:180]: ### After

https://github.com/MetaMask/metamask-extension/assets/8112138/6069db1c-2230-4b8e-8c4e-6819a90caf36

File: ui/components/app/snaps/snap-version/snap-version.js [hunk 5]
```diff
@@ -1,58 +1,33 @@
...
-          color={Color.textAlternative}
-          size={IconSize.Sm}
-          marginLeft={1}
-        />
-      </Box>
-    </Button>
+      {version ? version : <Preloader size={16} />}
+    </ButtonBase>
   );
 };
```
Comment (georgewrmarshall) [inline ui/components/app/snaps/snap-version/snap-version.js:33]: ### After


https://github.com/MetaMask/metamask-extension/assets/8112138/19eae9f1-2cf2-4fc8-ab09-990dfff44428

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Thanks for your contribution 🙏

**Review by georgewrmarshall — DISMISSED** [review]
Currently `develop` is failing CI so we need to wait for a fix but this LGTM

**Review by georgewrmarshall — DISMISSED** [review]

**Review by garrettbear — DISMISSED** [review]

**Review by georgewrmarshall — APPROVED** [review]

**Review by eriknson — APPROVED** [review]
LGTM!

**Comment by georgewrmarshall** [comment]
Would you mind fixing the linting issues @dhruvv173?

**Comment by dhruvv173** [comment]
> Would you mind fixing the linting issues @dhruvv173?

it was a bad merge on develop which was causing that lint error.

**Comment by georgewrmarshall** [comment]
Hey @dhruvv173, would you be able to look at the failing e2e test and update the selector?

**Comment by georgewrmarshall** [comment]
Hey @dhruvv173, would you be able to resolve the conflicts in this PR?

**Comment by georgewrmarshall** [comment]
Hey @dhruvv173, because this hasn't been worked on in a while I'm going to close this PR. Feel free to reopen it if you'd like to work on it again

**Pull request closed without merging**